src/repositories/upload.repository.ts

```diff
@@ -1,35 +1,30 @@
 import { random } from 'lodash';
 import { Repository } from '../core/repository';
-import { UploadPhotoOptions } from '../types/upload.photo.options';
 import Chance = require('chance');
-import { UploadRepositoryPhotoResponseRootObject } from '../responses';
-import { UploadVideoOptions } from '../types/upload.video.options';
+import { StatusResponse, UploadRepositoryPhotoResponseRootObject } from '../responses';
+import {
+  SEGMENT_DIVIDERS,
+  UploadRetryContext,
+  UploadSegmentedVideoOptions,
+  UploadVideoOptions,
+  UploadPhotoOptions,
+} from '../types';
 
 export class UploadRepository extends Repository {
```

Comment:
Sorry, tooooooo big Pull Request, i just couldn't check it out completely even in 2 calls.
I have one asking - if it is possible for you, could you please do different features in different branches and therefore create separate PR's, because it's too hard 😄 

This is the last notice i hope.

This whole class now violates the basic repository rule - one api-request per method call.

https://github.com/dilame/instagram-private-api/blob/master/src/repositories/README.md

Therefore we now have almost 100-lines `segmentedVideoUpload` function. I'm just trying to keep codebase simple

Reply:
I could fix it by myself, but there is too complex logic

Reply:
I'm sorry, I don't really PR this large amount of code and am not that experienced with them. Should I refactor it now into smaller PRs? (or at least the sticker service?)

Reply:
No, it's not important already since i had reviewed all the changes.